service/consul/datacenterWatch.go

```diff
@@ -1,109 +1,199 @@
 )
 
-//datacenterWatcher checks if datacenters have been updated, based on an interval
-//If they have, then the datacenterWatcher will update the Instancers dictionary
+//DatacenterWatcher checks if datacenters have been updated, based on an interval
 type datacenterWatcher struct {
-	watchInterval time.Duration
-	logger        log.Logger
-	shutdown      chan struct{}
-	environment   Environment
-	options       Options
+	logger              log.Logger
+	environment         Environment
+	options             Options
+	inactiveDatacenters map[string]bool
+	chrysomClient       *chrysom.Client
+	consulWatchInterval time.Duration
+	lock                sync.RWMutex
+}
+
+type datacenterFilter struct {
+	Name     string `mapstructure:"name"`
+	Inactive bool   `mapstructure:"inactive"`
 }
 
 var (
 	defaultLogger = log.NewNopLogger()
 )
 
 func newDatacenterWatcher(logger log.Logger, environment Environment, options Options) (*datacenterWatcher, error) {
-	if options.DatacenterWatchInterval == 0 {
-		return nil, errors.New("interval cannot be 0")
-	}
 
 	if logger == nil {
 		logger = defaultLogger
 	}
 
-	return &datacenterWatcher{
-		watchInterval: options.DatacenterWatchInterval,
-		logger:        logger,
-		shutdown:      make(chan struct{}),
-		environment:   environment,
-		options:       options,
-	}, nil
+	if options.DatacenterWatchInterval <= 0 {
+		//default consul interval is 5m
+		options.DatacenterWatchInterval = time.Duration(5 * time.Minute)
+	}
 
-}
+	datacenterWatcher := &datacenterWatcher{
+		consulWatchInterval: options.DatacenterWatchInterval,
+		logger:              logger,
+		options:             options,
+		environment:         environment,
+		inactiveDatacenters: make(map[string]bool),
+	}
 
-func (i *datacenterWatcher) start() {
-	go i.watchDatacenters()
-}
+	if options.ChrysomConfig != nil && options.ChrysomConfig.PullInterval > 0 {
 
-func (i *datacenterWatcher) stop() {
-	close(i.shutdown)
-}
+		if environment.Provider() == nil {
+			return nil, errors.New("must pass in a metrics provider")
+		}
 
-func (i *datacenterWatcher) watchDatacenters() {
+		options.ChrysomConfig.MetricsProvider = environment.Provider()
+
+		var datacenterListenerFunc chrysom.ListenerFunc = func(items []model.Item) {
+			updateInactiveDatacenters(items, datacenterWatcher.inactiveDatacenters, &datacenterWatcher.lock, logger)
+		}
 
-	environment := i.environment
-	client := i.environment.Client()
-	logger := i.logger
-	options := i.options
+		options.ChrysomConfig.Listener = datacenterListenerFunc
+
+		options.ChrysomConfig.Logger = logger
+		chrysomClient, err := chrysom.CreateClient(*options.ChrysomConfig)
+
+		if err != nil {
+			return nil, err
+		}
 
-	checkDatacenters := time.NewTicker(i.watchInterval)
+		//create chrysom client and start it
+		datacenterWatcher.chrysomClient = chrysomClient
+		datacenterWatcher.chrysomClient.Start(context.Background())
 
+	} else if options.ChrysomConfig != nil && options.ChrysomConfig.PullInterval <= 0 {
```

Comment:
I think moving this case earlier will make the code easier to understand.

Instead of having:
```
if config != nil && pullInterval > 0 {
...
} else if config != nil && pullInterval <= 0 {
return
}
...
```

We can have something like this:
```
if config != nil {
    if pullInterval <= 0 {
        return 
    }
    ...
}
...
```

It would fit in with the nil check for the `environment.Provider()` as well.